tests/MongoDB.Driver.Tests/Specifications/transactions-convenient-api/TransactionsConvenientApiTestRunner.cs

```diff
@@ -0,0 +1,90 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Tests.Specifications.Runner;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.transactions_convenient_api
+{
+    public class TransactionsConvenientApiTestRunner : MongoClientJsonDrivenSessionsTestRunner
+    {
+        [SkippableTheory]
+        [ClassData(typeof(TestCaseFactory))]
+        public void Run(JsonDrivenTestCase testCase)
+        {
+            RunTests(testCase);
+        }
+
+        protected override void CheckServerRequirements(BsonDocument document)
+        {
+            base.CheckServerRequirements(document);
+            var clusterType = CoreTestConfiguration.Cluster.Description.Type;
+            if (clusterType == ClusterType.ReplicaSet)
+            {
+                RequireServer.Check().Supports(Feature.WithTransactionReplicaSet);
+            }
+            else if (clusterType == ClusterType.Sharded)
+            {
+                RequireServer.Check().Supports(Feature.WithTransactionShardedCluster);
+            }
+            else
+            {
```

Comment:
I didn't find in spec that this logic should be implemented only for ReplicaSet or ShardedCluster.

>The only one which I found is:
   Testing against a replica set will require server version 4.0.0 or later. The replica set should include a primary, a secondary, and an arbiter. Including a secondary ensures that server selection in a transaction works properly. Including an arbiter helps ensure that no new bugs have been introduced related to arbiters.

>Testing against a sharded cluster will require server version 4.1.6 or later. A driver that implements support for sharded transactions MUST also run these tests against a MongoDB sharded cluster with multiple mongoses. Including multiple mongoses (and initializing the MongoClient with multiple mongos seeds!) ensures that mongos transaction pinning works properly.

I don't see anything about standalone servers. However, unit tests on all standalone servers are faied. Also, `TransactionTestRunner` allows only replica set. So, I assume that these tests should be run only on shurded cluster and replica set.